tests/autoValue.tests.js

```diff
@@ -1,91 +1,106 @@
 
 [collection, localCollection].forEach((c) => {
   c.attachSchema(new SimpleSchema({
     clientAV: {
       type: SimpleSchema.Integer,
       optional: true,
-      autoValue() {
-        if (Meteor.isServer) return;
-        return (this.value || 0) + 1;
+      autoValue () {
+        if (Meteor.isServer) return
+        return (this.value || 0) + 1
       }
     },
     serverAV: {
       type: SimpleSchema.Integer,
       optional: true,
-      autoValue() {
-        if (Meteor.isClient) return;
-        return (this.value || 0) + 1;
+      autoValue () {
+        if (Meteor.isClient) return
+        return (this.value || 0) + 1
       }
     }
-  }));
-});
+  }))
+})
 
 if (Meteor.isClient) {
   describe('autoValue on client', function () {
     it('for client insert, autoValues should be added on the server only (added to only a validated clone of the doc on client)', function (done) {
       collection.insert({}, (error, id) => {
-        const doc = collection.findOne(id);
-        expect(doc.clientAV).toBe(undefined);
-        expect(doc.serverAV).toBe(1);
-        done();
-      });
-    });
+        if (error) {
+          done(error)
+          return
+        }
+        const doc = collection.findOne(id)
+        expect(doc.clientAV).toBe(undefined)
+        expect(doc.serverAV).toBe(1)
+        done()
+      })
+    })
 
     it('runs function once for LocalCollection', function (done) {
       localCollection.insert({}, (error, id) => {
-        const doc = localCollection.findOne(id);
-        expect(doc.clientAV).toBe(1);
-        expect(doc.serverAV).toBe(undefined);
-        done();
-      });
-    });
+        if (error) {
+          done(error)
+          return
+        }
+        const doc = localCollection.findOne(id)
+        expect(doc.clientAV).toBe(1)
+        expect(doc.serverAV).toBe(undefined)
+        done()
+      })
+    })
 
     it('with getAutoValues false, does not run function for LocalCollection', function (done) {
       localCollection.insert({}, { getAutoValues: false }, (error, id) => {
-        const doc = localCollection.findOne(id);
-        expect(doc.clientAV).toBe(undefined);
-        expect(doc.serverAV).toBe(undefined);
-        done();
-      });
-    });
-  });
+        if (error) {
+          done(error)
+          return
+        }
+        const doc = localCollection.findOne(id)
+        expect(doc.clientAV).toBe(undefined)
+        expect(doc.serverAV).toBe(undefined)
+        done()
+      })
+    })
+  })
 }
 
 if (Meteor.isServer) {
   describe('autoValue on server', function () {
-    it('runs function once', function () {
-      const id = collection.insert({});
-      const doc = collection.findOne(id);
-      expect(doc.clientAV).toBe(undefined);
-      expect(doc.serverAV).toBe(1);
-    });
+    it('runs function once', async function () {
+      const id = await callMongoMethod(collection, 'insert', [{}])
```

Comment:
How can we ensure if it's backwards compatible if all the tests now use the async version? Shouldn't we keep the old sync test and also async ones? 
This is a question really as I'm new to the async lands :sweat_smile: